provisioning/miniprov/hare_mp/main.py

```diff
@@ -254,21 +255,202 @@ def test(args):
         exit(-1)
 
 
+def executecmds(cmd: List[str]) -> list:
+    process = subprocess.Popen(cmd,
+                               stdin=subprocess.PIPE,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               encoding='utf8')
+    resp = list(process.communicate())
+    resp.append(str(process.returncode))
+
+    return resp
+
+
+def test_hare_prereq():
+    """Test suite for setting the test environment before start of test."""
+    pcs_status = ['pcs', 'status']
+    cluster_stop = ['cortx', 'cluster', 'stop']
+
+    logging.info('Test started on Host: {}'.format(executecmds(['hostname'])))
+    logging.info('Check that all services are up in PCS.')
+
+    resp = executecmds(pcs_status)
+    logging.info('PCS status: %s', resp[0])
+
+    if 'cluster is not currently' in resp[0]:
+        return
+
+    logging.info('Make Node ready for testing, by stopping the cluster')
+    resp = executecmds(cluster_stop)
+    if int(resp[2]):
+        logging.info('Cluster status : %s', resp[0])
+        logging.error('Cluster failed to stop %s', resp[1])
+        raise Exception(
+            f'Command {cluster_stop} exited with error code '
+            f'{int(resp[2])}. Command output: {resp[1]}')
+    else:
+        logging.info('Cluster Stopped : %s', resp[0])
+        for line in resp[0]:
```

Comment:
This deserves a separate function. And please think how to avoid `assert` instruction.